compiler-core/src/parse.rs

```diff
@@ -3486,6 +3489,19 @@ fn is_reserved_word(tok: Token) -> bool {
     }
 }
 
+/// Formats a token into a debug-style string and specifies whether
+/// the token is a reserved word (i.e., keyword).
+fn format_found_token(tok: Token) -> String {
+    let base = tok.to_string();
+    if is_reserved_word(tok) {
+        format!("{base} (a keyword)")
+    }
+    else {
+        format!("{base}")
+    }
+}
```

Comment:
This is error presentation logic so it needs to live in the warning printer, and the data in the warning needs to be designed data structures rather than strings. Perhaps it could contain the token itself.